soql-standalone-parser/src/main/scala/com/socrata/soql/parsing/AbstractParser.scala

```diff
@@ -95,10 +95,15 @@ abstract class AbstractParser extends Parsers with PackratParsers {
   def pipedSelect: Parser[Seq[Select]] = rep1sep(unchainedSelect, QUERYPIPE())
 
   def unchainedSelect: Parser[Select] =
-    SELECT() ~> selectList ~ opt(whereClause) ~ opt(groupByClause) ~ opt(havingClause) ~ orderByAndSearch ~ limitOffset ^^ {
-      case s ~ w ~ gb ~ h ~ ((ord, sr)) ~ ((lim, off)) => Select(s, w, gb, h, ord, lim, off, sr)
+    SELECT() ~> distinct ~ selectList ~ opt(whereClause) ~ opt(groupByClause) ~ opt(havingClause) ~ orderByAndSearch ~ limitOffset ^^ {
+      case d ~ s ~ w ~ gb ~ h ~ ((ord, sr)) ~ ((lim, off)) => Select(d, s, w, gb, h, ord, lim, off, sr)
     }
 
+  def distinct: Parser[Boolean] = opt(DISTINCT()) ^^ {
+    case Some(_) => true
+    case None => false
+  }
+
```

Comment:
`opt(DISTINCT()) ^^ (_.isDefined)` would be shorter.  I'm not insistent upon it if you like this way better though.